src/anyp/TrafficMode.h

```diff
@@ -29,15 +24,29 @@ class TrafficMode
      */
     bool accelSurrogate = false;
 
-    /** marks ports receiving PROXY protocol traffic
+    /** marks http ports receiving PROXY protocol traffic
```

Comment:
If I understand Eduard's explanation correctly, if some future code adds PROXY protocol support for a port foo (other than http_port and https_port), that code may need to add a proxySurrogateFoo flag to prevent breaking existing proxySurrogateHttp users that rely on the current http_port association. @eduard-bagdasaryan, can you point to any existing PR code that relies on that association? If there is no such code, then Amos' underlying concern is valid (the code changes to address that concern would look different, but that is not important for now).

Reply:
> any existing PR code that relies on that association

No, there is no such association. I think we can use this proxySurrogateHttp for another port, e.g, ftp_port - as soon as the required functionality is provided. The description I suggested reflects the current status, not a hypothetical one that would emerge in the future. I do not insist on this descriptive 'http', though.

Reply:
To be clearer on what I want to see here:

* config options only set one flag, regardless of which port type they happen on.
* use an accessor method or flags.foo&&flags.bar to perform the check

That means this PR should add proxySurrogateHttp() { return proxy Surrogate /* && http_port */ ; } and proxySurrogateHttps() {return proxySurrogate && sslBump; /* && https_port*/ ; }
They may need to be method's of PortCfg to get the port protocol type/scheme

Reply:
> They may need to be method's of PortCfg to get the port protocol type/scheme

IIRC I considered the approach you suggest during development but encountered some difficulties. To begin with, TrafficModeFlags is designed as a simple POD structure without any references - it does not know about PortCfg. Even if we would add this reference to TrafficModeFlags (creating a circular dependency!), there is no such information as raw 'port type' (i.e, http_port or https_port) stored in PortCfg which can be used to distinguish between proxySurrogateHttp() and proxySurrogateHttps(). It has PortCfg::transport field (configured from the port type) containing ProtocolVersion, but it is not obvious how to use ProtocolVersion in the suggested methods.

Reply:
> Amos: To be clearer on what I want to see here:
> * config options only set one flag, regardless of which port type they happen on.
> * use an accessor method or flags.foo&&flags.bar to perform the check of any permutations

I think it is possible to fulfill Amos desires quoted above by:

* adding one new raw data member to store the port kind/type (we cannot use PortCfg::transport because it gets overwritten by `protocol=`);
* renaming and adjusting tunnelSslBumping maintenance so that it is never reset to false by non-configuration code and, hence, can be used for determining the presence of `ssl-bump` option in http_port. Hopefully, we can find a way to check ssl_bump rules presence quickly where needed. These SslBump adjustments are out of this PR scope, but I see no way to both satisfy Amos' requirements without enlarging the scope of this work.

@yadij, will you accept the above solution?

----

> `proxySurrogateHttps() { return proxySurrogate && sslBump && this-is-https_port; }`

Yes, except we should not use that overreaching method name to avoid the confusion that wasted a lot of time already, but this is minor.

Reply:
> fulfill Amos desires

Done (up to cd55635).